src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -291,8 +297,147 @@ OCC::Optional<OCC::PinStateEnums::PinState> OCC::CfApiWrapper::PlaceHolderInfo::
     return cfPinStateToPinState(_data->PinState);
 }
 
-OCC::Result<void, QString> OCC::CfApiWrapper::registerSyncRoot(const QString &path, const QString &providerName, const QString &providerVersion)
+QString convertSidToStringSid(void *sid)
+{
+    QString result;
+
+    wchar_t *stringSid = nullptr;
+    if (::ConvertSidToStringSid(sid, &stringSid)) {
+        result = QString::fromWCharArray(stringSid);
+    }
+
+    if (stringSid) {
+        ::LocalFree(stringSid);
+        stringSid = nullptr;
+    }
+
+    return result;
+}
+
+QString retrieveWindowsSID()
+{
+    // FYI: code here is mostly identical to Windows-Classic-Examples/Samples/CloudMirror
+    QScopedPointer<TOKEN_USER> tokenInfo;
+
+    const auto tokenHandle = GetCurrentThreadEffectiveToken();
+
+    DWORD tokenInfoSize = { 0 };
+    // first - get the required size
+    if (!::GetTokenInformation(tokenHandle, TokenUser, nullptr, 0, &tokenInfoSize))
```

Comment:
In fact this should always fail, otherwise we're in big trouble. :-)

I think I'd go for (man... I hate this API):
```cpp
const auto tokenSizeCallSucceeded = GetTokenInformation(tokenHandle, TokenUser, nullptr, 0, &tokenInfoSize);
const auto lastError = GetLastError;
Q_ASSERT(!tokenSizeCallSucceeded && lastError == ERROR_INSUFFICIENT_BUFFER);
if (tokenSizeCallSucceeded || lastError != ERROR_INSUFFICIENT_BUFFER) {
    // Log it
    return {};
}

tokenInfo.reset(reinterpret_cast<TOKEN_USER*>(new char[tokenInfoSize]));
if (!GetTokenInformation(tokenHandle, TokenUser, tokenInfo.get(), tokenInfoSize, &tokenInfoSize)) {
    // log it
    return {};
}

return convertSidToStringSid(tokenInfo->User.Sid);
```

Something along those lines...

Reply:
@er-vin Ok. Done.